### PR TITLE
Added functionality for uploading files

### LINKED ISSUE #616: [Feature] Implement uploading campaign files
**Is your feature request related to a problem? Please describe.**
We need to support uploading files for a campaign. Currently, there is an upload dialog from uppy that is integrated in our frontend. However, we chose not to use Uppy as it is too heavy for our simple use-case. The back-end for uploading files is already implemented and deployed to development. You can see the changes [here](https://github.com/podkrepi-bg/api/pull/123)

**Describe the solution you'd like**
The component from Uppy should be removed and we should have a simpler solution that just sends the selected files to the API. We can do it simple at the beginning by just listing the file names that are going to be uploaded. However, in the future there will be a need to have a better visualization. I only managed to find:
- [component 1](https://react-dropzone-uploader.js.org/)
- [component 2](https://www.positronx.io/react-image-upload-with-preview-percentage-progress-bar-tutorial/)

**Additional context**
The current implementation of the back-end expects a multipart form-data to be submitted with the files. All files should be indexed with the `file` key. It is important to make sure the user's authentication token is passed to the back-end when uploading since the uploader is taken from the token itself.

### PR DESCRIPTION
**Pull request by dplamenov**
Added functionality for uploading files when creating a campaign.

Screenshot at Mar 22 15-16-11
Screenshot at Mar 22 15-16-28

 Should close #616

### CODE REVIEW

File: src/components/campaigns/CampaignForm.tsx [hunk 1]
```diff
@@ -225,7 +239,23 @@ export default function CampaignForm({ initialValues = defaults }: CampaignFormP
...
+              />
+              <Button fullWidth variant="contained" color="info" component="span">
+                Добави снимки
+              </Button>
+            </label>
+            <ul>
+              {files.map((file) => {
+                return <li key={file.name}>{file.name}</li>
+              })}
+            </ul>
```
[the remark this answers is not in the document]
Comment (kachar) [reply src/components/campaigns/CampaignForm.tsx:258]: The list of uploaded files can be a separate component

[the remark this answers is not in the document]
Comment (dplamenov) [reply src/components/campaigns/CampaignForm.tsx:258]: Ok.

[the remark this answers is not in the document]
Comment (kachar) [reply src/components/campaigns/CampaignForm.tsx:258]: Something similar to that

```tsx
export default function CreateForm() {
  const [files, setFiles] = useState<File[]>([])
  return (
    <FileUploadModal onUpload={file => setFiles(files => [...files, file])} />
  )
}
```

In this way the uploading mechanism is encapsulated in `fileuploadmodal` and the state of the uploaded files is part of the campaign form.

[the remark this answers is not in the document]
Comment (dplamenov) [reply src/components/campaigns/CampaignForm.tsx:258]: > Something similar to that
> 
> ```tsx
> export default function CreateForm() {
>   const [files, setFiles] = useState<File[]>([])
>   return (
>     <>
>       <FileUploadModal onUpload={file => setFiles(files => [...files, file])} />
>       <FileList files={files} />
>     </>
>   )
> }
> ```
> 
> In this way the uploading mechanism is encapsulated in `fileuploadmodal` and the state of the uploaded files is part of the campaign form.

Ok. I will do ASAP

**Review by kachar — APPROVED** [review]
@dplamenov Cool upgrade of the app, really needed for the next steps of the campaign creation.

Thanks for your contribution :rocket:

**Comment by kachar** [comment]
image
Added MUI list component to the list of files with a delete button for each file

**Pull request merged**